pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, pkgs
+, runtimeShell
+, symlinkJoin
+, tree
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
+      in
+        let p = callPackage datum.expressionPath parameters;
+        in acc // { ${datum.name} = p; };
+    in builtins.foldl' _packageWithLinkCommandFrom {} _wordlistData
+  );
```

Comment:
The only thing I don't really like about this PR is this big expression, which can be rather easily simplified.

Reply:
@zseri This kind of review is not helpful at best.

I have tried my best to make this expression as elegant in nix while still respecting the way I wanted to use it.
For that, I had to dig into the documentation, and bang my head at walls to try to follow advice that was given along the way (I did not know everything right of the bat).
It's fine, I was happy to do it, I learned stuff along the way, and I understand there are quality requirements for my work to match the expectations and needs of the community.

> rather easily simplified

Please enlighten us...
Sarcasm aside, the use of "easily" is unnecessary, and can be quite upsetting for those of us who do not find that "easy" (https://css-tricks.com/words-avoid-educational-writing/#10-easy).

So, unless there is something specific (and pertinent) that can be pointed out, and some kind of guidance about how to fix it: :shrug:

Reply:
ok, sorry for my bluntness... here's a patch: https://github.com/zseri/nixpkgs/commit/3667eb616bbd51ea7ee1318c09317a38e85c86e2.
Tested with: `nix-build -I nixpkgs=./. ../nixpkgs-wordlists-test.nix`
```nix
# ../nixpkgs-wordlists-test.nix
{ nixpkgs ? <nixpkgs>, system ? builtins.currentSystem }:
(import nixpkgs { inherit system; }).pkgs.security-wordlists.withLists (ps: with ps; [ dirbuster nmap ])
```

Reply:
Thanks for sharing this!
Indeed, this looks much less convoluted.